src/dwn/models/record.js

```diff
@@ -0,0 +1,244 @@
+    const mutableDescriptorProperties = ['dataCid', 'dataSize', 'dateModified', 'datePublished', 'published'];
+    Record.#verifyPermittedMutation(Object.keys(options), mutableDescriptorProperties);
+
+    // If a new `dateModified` was not provided, remove it from the updateMessage to let the DWN SDK auto-fill.
+    // This is necessary because otherwise DWN SDK throws an Error 409 Conflict due to attempting to overwrite a record
+    // when the `dateModified` timestamps are identical.
+    if (options?.dateModified === undefined) {
+      delete updateMessage.dateModified;
+    }
+
+    // If `published` is set to false, ensure that `datePublished` is undefined. Otherwise, DWN SDK's schema validation
+    // will throw an error if `published` is false but `datePublished` is set.
+    if (options?.published === false && updateMessage?.datePublished !== undefined) {
+      delete updateMessage.datePublished;
+    }
+
+    // Set the record ID and context ID, if any.
+    updateMessage.recordId = this.#recordId;
+    updateMessage.contextId = this.#contextId;
+
+    // Attempt to write the changes to mutable properties to the DWN.
+    const { message = null, record = null, status } = await this.#dwn.records.write(this.#target, {
+      author: this.#author,
+      data,
+      message: {
+        ...updateMessage,
+      },
+    });
+
+    if (status.code === 202 && record) {
+      // Only update the local Record instance mutable properties if the record was successfully (over)written.
+      mutableDescriptorProperties.forEach(property => {
+        this.#descriptor[property] = record[property];
+      });
+      // Only update data if `dataSize` is less than DWN 'max data size allowed to be encoded'.
+      if (data !== undefined) {
+        this.#readableStream = (record.dataSize <= DwnConstant.maxDataSizeAllowedToBeEncoded) ? record.data : null;
+        this.#encodedData = null; // Clear `encodedData` in case it was previously set.
+      }
+    }
+
+    return { message, status };
+  }
+
+  /**
+   * Utility methods.
+   */
+
+  /**
+   * Called by `JSON.stringify(...)` automatically.
+   */
+  toJSON() {
+    return {
+      author: this.author,
+      target: this.target,
+      recordId: this.id,
+      contextId: this.contextId,
+      dataFormat: this.dataFormat,
+      dateCreated: this.dateCreated,
+      interface: this.interface,
+      method: this.method,
+      parentId: this.parentId,
+      protocol: this.protocol,
+      recipient: this.recipient,
+      schema: this.schema,
+      dataCid: this.dataCid,
+      dataSize: this.dataSize,
+      dateModified: this.dateModified,
+      datePublished: this.datePublished,
+      published: this.published,
+    };
+  }
+
+  #freezeRecord() {
+    this.#isFrozen = true;
+  }
+
+  #unFreezeRecord() {
+    this.#isFrozen = false;
+  }
+
+  static #verifyPermittedMutation(propertiesToMutate, mutableDescriptorProperties) {
+    propertiesToMutate.forEach(propertyName => {
```

Comment:
id use `for (const property of propertiesToMutate)` as that would allow for this to be called with an array or `Set` or etc. (i.e. any flat iterable) instead of just an array

Reply:
Since its a private method, we know and control the type for `propertiesToMutate()`.  However, sure... its more flexible if we ever want to use the method for other purposes.

Addressed in 96c7a8030cff76db65305578bc3beb34ffc6fc51